controllers/components/kserve/kserve_controller.go

```diff
@@ -19,40 +19,133 @@ package kserve
 
 	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	featuresv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/features/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/gc"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/render/kustomize"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/security"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/updatestatus"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/predicates/clusterrole"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/predicates/hash"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/predicates/resources"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/reconciler"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
 )
 
-// KserveReconciler reconciles a Kserve object.
-type KserveReconciler struct {
-	client.Client
-	Scheme *runtime.Scheme
-}
+// NewComponentReconciler creates a ComponentReconciler for the Dashboard API.
+func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.Manager) error {
+	ownedViaFTMapFunc := ownedViaFT(mgr.GetClient())
 
-// Reconcile is part of the main kubernetes reconciliation loop which aims to
-// move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the Kserve object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
-func (r *KserveReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
-
-	// TODO(user): your logic here
-
-	return ctrl.Result{}, nil
-}
+	_, err := reconciler.ComponentReconcilerFor(mgr, componentsv1.KserveInstanceName, &componentsv1.Kserve{}).
+		// operands - owned
+		Owns(&corev1.Secret{}).
+		Owns(&corev1.Service{}).
+		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.ServiceAccount{}).
+		Owns(&rbacv1.Role{}).
+		Owns(&rbacv1.RoleBinding{}).
+		Owns(&rbacv1.ClusterRole{}, reconciler.WithPredicates(clusterrole.IgnoreIfAggregationRule())).
+		Owns(&rbacv1.ClusterRoleBinding{}).
+		// The ovms template gets a new resourceVersion periodically without any other
+		// changes. The compareHashPredicate ensures that we don't needlessly enqueue
+		// requests if there are no changes that we don't care about.
+		Owns(&templatev1.Template{}, reconciler.WithPredicates(hash.Updated())).
+		Owns(&featuresv1.FeatureTracker{}).
```

Comment:
Even though I've got this `Owns` set for FeatureTrackers here, I can't seem to get updates to a FeatureTracker in the cluster (which has the correct owner reference set), to cause a new reconciliation, and I don't understand why. Thoughts?

Reply:
I think it depends what changes, if the spec does not changes, then nothing would probably happen.
If you need the status to be watched, then a custom set of predicated have to be configured.

Reply:
uff, I tough I solved this, have to digg into that

Reply:
It seems like always happens: there's nothing else happening and no matter how many times I make a change to the CR, no reconcile is triggered for it. It sounds like you've experienced it before...any idea what it might be?

Reply:
Thanks! I'm not so worried about FeatureTrackers, since it seems to just be a kind of marker for the controller that creates them (the kserve controller in this case), but it would be good to make sure that it's not liable to happen for more important resources.

Also, it's very likely/possible that I've just missed something obvious!

Reply:
Interesting, thanks!

This seems to be because the Feature stuff uses [controllerutil.SetOwnerReference()](https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/controller/controllerutil/controllerutil.go#L110) rather than [controllerutil.SetControllerReference()](https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/controller/controllerutil/controllerutil.go#L70).

Since FeatureTrackers are used in other places too (DSCI?), I'll try not to change that in case in breaks somewhere else. Instead I'll first see if I can create a custom eventHandler that will work here.

Reply:
Just saw the comment now, but yes, I came to the same conclusion and have updated here now, and it works as expected. Thanks!